<launch>
  <!-- valid option:
        mask_rcnn_X_101_32x8d_FPN_3x
        mask_rcnn_R_101_FPN_3x
        mask_rcnn_R_101_DC5_3x
        mask_rcnn_R_101_C4_3x
        mask_rcnn_R_50_FPN_3x -->

  <!-- ============================================================================= -->
  <!-- |               name               | inference time [s/im] | boxAP | maskAP | -->
  <!-- ============================================================================= -->
  <!-- | mask_rcnn_X_101_32x8d_FPN_3x     | 0.103                 | 44.3  | 39.5   | -->
  <!-- ============================================================================= -->
  <!-- | mask_rcnn_R_101_FPN_3x           | 0.056                 | 42.9  | 38.6   | -->
  <!-- ============================================================================= -->
  <!-- | mask_rcnn_R_101_DC5_3x           | 0.092                 | 41.9  | 37.3   | -->
  <!-- ============================================================================= -->
  <!-- | mask_rcnn_R_101_C4_3x            | 0.145                 | 42.6  | 36.7   | -->
  <!-- ============================================================================= -->
  <!-- | mask_rcnn_R_50_FPN_3x            | 0.043                 | 41.0  | 37.2   | -->
  <!-- ============================================================================= -->

  <!-- model settings -->
  <arg name="base_model_name"                 default="mask_rcnn_R_101_FPN_3x" />
  <arg name="custom_model_name"               default="mask_rcnn_R_101_FPN_3x_iter_9000_learn_5e-05" />
  <arg name="model"                           default="$(find detectron2_ros)/detectron2/configs/COCO-InstanceSegmentation/$(arg base_model_name).yaml" />
  <arg name="base_model_weight"               default="$(find detectron2_ros)/detectron2/weights/COCO-InstanceSegmentation/$(arg base_model_name).pkl" />
  <arg name="custom_model_weight"             default="$(find detectron2_ros)/detectron2/weights/COCO-InstanceSegmentation/$(arg custom_model_name).pth" />

  <!-- other parameters -->
  <arg name="detection_threshold"             default="0.6" />
  <arg name="generate_image_from_pcd"         default="false" />
  <arg name="use_compressed_image"            default="false" />
  <arg name="visualization"                   default="true" />
  <arg name="publish_rate"                    default="100" />
  <arg name="enable_quater_pcd"               default="false" />
  <arg name="use_custom_model"                default="false" />
  <arg name="cam_index"                       default="99" />

  <node pkg="detectron2_ros" type="detectron2_ros.py" name="detectron2_ros_$(arg cam_index)" output="screen"><!--  output="screen" -->
    <param name="cam_index"                       value="$(arg cam_index)" />
    <param name="detection_threshold"             value="$(arg detection_threshold)" />
    <param name="model"                           value="$(arg model)" />
    <param name="base_model_weight"               value="$(arg base_model_weight)" />
    <param name="custom_model_weight"             value="$(arg custom_model_weight)" />
    <!-- <param name="weight"                          value="$(arg weight)" /> -->
    <param name="visualization"                   value="$(arg visualization)" />
    <param name="publish_rate"                    value="$(arg publish_rate)" />
    <param name="generate_image_from_pcd"         value="$(arg generate_image_from_pcd)" />
    <param name="use_compressed_image"            value="$(arg use_compressed_image)" />
    <param name="enable_quater_pcd"               value="$(arg enable_quater_pcd)" />
    <param name="use_custom_model"                value="$(arg use_custom_model)" />
    <remap from="/env_cam99/rgb/image_raw"      to="/generated_img" />
  </node>
</launch>